src/Serilog.Sinks.EventLog/Sinks/EventLog/EventLogSink.cs

```diff
@@ -59,23 +47,33 @@ public EventLogSink(string source, string? logName, ITextFormatter textFormatter
     /// <param name="machineName">The name of the machine hosting the event log written to.</param>
     /// <param name="manageEventSource">If false does not check/create event source.  Defaults to true i.e. allow sink to manage event source creation</param>
     /// <param name="eventIdProvider">Supplies event ids for emitted log events.</param>
-    public EventLogSink(string source, string? logName, ITextFormatter textFormatter, string machineName, bool manageEventSource, IEventIdProvider eventIdProvider)
+    /// <param name="categoryNumberProvider">Supplies category numbers for emitted log events.</param>
+    public EventLogSink(string source, string? logName, ITextFormatter textFormatter, string machineName, bool manageEventSource, IEventIdProvider? eventIdProvider = null, ICategoryNumberProvider? categoryNumberProvider = null)
     {
         if (source == null) throw new ArgumentNullException(nameof(source));
         if (textFormatter == null) throw new ArgumentNullException(nameof(textFormatter));
-        if (eventIdProvider == null) throw new ArgumentNullException(nameof(eventIdProvider));
+        if (eventIdProvider == null)
```

Comment:
Instead of reassigning the parameters, it might be clearer to just use null coalescing for the assignments below -

```csharp
_eventIdProvider = eventIdProvider ?? new EventIdHashProvider();
```

Reply:
Good point - replaced with null coalescing